src/FsAutoComplete/CodeFixes.fs

```diff
@@ -0,0 +1,719 @@
+        | CoreResponse.Res (text, position) ->
+            let replacements = getTextReplacements ()
+
+            let replaced =
+              (text, replacements)
+              ||> Seq.fold (fun text (KeyValue (key, replacement)) -> text.Replace(key, replacement))
+
+            return
+              [ { SourceDiagnostic = None
+                  Title = "Generate interface stub"
+                  File = codeActionParams.TextDocument
+                  Edits =
+                    [| { Range = fcsPosToProtocolRange pos
+                         NewText = replaced } |] } ]
+        | _ -> return []
+
+      }
+      |> AsyncResult.foldResult id (fun _ -> [])
+
+  /// a codefix that generates member stubs for a record declaration
+  let generateRecordStub (getFileLines: string -> Result<string [], _>)
+                         (getParseResultsForFile: string -> FSharp.Compiler.Range.pos -> Async<Result<ParseAndCheckResults * string * string array, 'e>>)
+                         (genRecordStub: _ -> _ -> _ -> _ -> Async<CoreResponse<string * FSharp.Compiler.Range.pos>>)
+                         (getTextReplacements: unit -> Map<string, string>)
+                         : CodeFix =
+    fun codeActionParams ->
+      asyncResult {
+        let fileName =
+          codeActionParams.TextDocument.GetFilePath()
+
+        let! (lines: string []) = getFileLines fileName
+
+        let pos =
+          protocolPosToPos codeActionParams.Range.Start
+
+        let! (tyRes, line, lines) = getParseResultsForFile fileName pos
+
+        match! genRecordStub tyRes pos lines line with
+        | CoreResponse.Res (text, position) ->
+            let replacements = getTextReplacements ()
+
+            let replaced =
+              (text, replacements)
+              ||> Seq.fold (fun text (KeyValue (key, replacement)) -> text.Replace(key, replacement))
+
+            return
+              [ { SourceDiagnostic = None
+                  Title = "Generate record stub"
+                  File = codeActionParams.TextDocument
+                  Edits =
+                    [| { Range = fcsPosToProtocolRange pos
+                         NewText = replaced } |] } ]
+        | _ -> return []
+      }
+      |> AsyncResult.foldResult id (fun _ -> [])
+
+  /// a codefix that adds in missing '=' characters in type declarations
+  let addMissingEqualsToTypeDefinition (getFileLines: string -> Result<string [], _>) =
+    ifDiagnosticByCode
+      (fun diagnostic codeActionParams ->
+        asyncResult {
+          if diagnostic.Message.Contains "Unexpected symbol '{' in type definition"
+             || diagnostic.Message.Contains "Unexpected keyword 'member' in type definition" then
+            let fileName =
+              codeActionParams.TextDocument.GetFilePath()
+
+            let! lines = getFileLines fileName
+
+            match walkBackUntilCondition lines (dec lines diagnostic.Range.Start) (System.Char.IsWhiteSpace >> not) with
```

Comment:
here we start walking back from the position before the start of the diagnostic because the diagnostic range covers the text of the `{`, which would satisfy the walk condition.